tracing,execinfrapb: include child metadata in trace agg metas

dt opened the pull request:
This captures and includes in aggregated trace info the "child span metadata", i.e. the  occurrence count count and sum duration of each uniquely named child span. These are included in the "per-component" and "cluster wide" trace dumps utilized by various jobs' advance debugging tools.

Release note: none.
Epic: none.

dt committed:
tracing,execinfrapb: include child metadata in trace agg metas

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143090)
<!-- Reviewable:end -->

msbutler commented on pkg/util/tracing/span_inner.go:
```diff
@@ -65,6 +65,12 @@ func (s *spanInner) GetTraceRecording(recType tracingpb.RecordingType, finishing
 
+// GetFullTraceRecording returns the span's full recording, including detached
+// children, as a Trace. See GetTraceRecording and WithDetachedRecording.
+func (s *spanInner) GetFullTraceRecording(recType tracingpb.RecordingType) Trace {
```
i don't understand why we don't need to check if `s.crdb != nil` but there's precedent for skipping that check.

I suppose all spans started by a `Tracer` are crdb spans.

dt replied:
Yeah, this is just following the precedent of the other ones. 🤷

msbutler commented on pkg/backup/backup_job.go:
```diff
@@ -513,6 +514,7 @@ type backupResumer struct {
 		// driven AggregatorStats pushed backed to the resumer from all the
 		// processors running the backup.
 		perNodeAggregatorStats bulkutil.ComponentAggregatorStats
+		perSpanMetadata        map[string]tracingpb.OperationMetadata
```
stray addition?

msbutler reviewed (approved):
Do you have a before/after job trace which illustrates the the additional metadata captured by this change?

dt committed:
jobs: reformat dumped trace stats

Previously we had one file that had cluster-wide and per-node stats.
It dumped per-node first, and cluster-wide after, making it harder to find. It also dumped the nodes in random (map) order.

Instead, we now dump the cluster-wide stats to their own file, making
these easy to find/consume, and the per-node stats are dumped in node
order, making them easier to browse as well. The formatting is also now
tweaked to make the sections clearer.

Finally the files are renamed to be timestamp-then-name, so that each time one
dumps stats, all the dumped files for that request sort togeher, vs sorting by type of file first.

Release note: none.
Epic: none.

dt committed:
jobs: dump debug files date first

Previously we suffixed files with the time they were made. This meant
that if you requested new files, you needed do look through all the files, including the old ones, to find those your request made. Instead, all files produced by one request now sort together.

Release note: none.
Epic: none.

dt commented:
Added another change to clean up the text dump of these stats, splitting it into two files for cluster-wide vs node-by-node. While I'm here, I'm also nuking the per-event-type raw proto files that clutter up the debug page and were never used, and renaming everything to be date-then-name instead of name-then-date, so the files dumped by each request should be grouped on the sorted page.

dt commented:
Here's the result:
<img width="1287" alt="Screenshot 2025-03-26 at 17 55 31" src="https://github.com/user-attachments/assets/3313267f-f854-44cb-b025-f72977ff5c38" />
[cluster-wide.txt](https://github.com/user-attachments/files/19475974/cluster-wide.txt)
[by-node.txt](https://github.com/user-attachments/files/19475973/by-node.txt)

dt commented:
and the cluster-wide and per-nodes: 
![Uploading Screenshot 2025-03-26 at 18.06.38.png…]()

msbutler reviewed (approved):
I gotta make sure @adityamaru sees these beautiful improvements to job observability
